File: Cuke4Duke/trunk/NCP_Cuke4Duke/src/main/java/com/sqs/nochangeparking/core/PaymentsList.java
package com.sqs.nochangeparking.core;

import java.util.ArrayList;

public class PaymentsList extends ArrayList<Payment>{
	
	private static final long serialVersionUID = 1L;


//	private PaymentsList():base(new List<Payment>()){/*NOOP*/}

	public boolean add(String vehicle) {
		if (contains(vehicle))
			return false;
		return super.add(new Payment(vehicle));
	}
	
	public boolean contains(String vehicle) {
		return contains(new Payment(vehicle));
	}
	
	public boolean hasPaid(String vehicle) {
		return contains(new Payment(vehicle));
	}
	
	public void AddPayment(String vehicle) {
		if (hasPaid(vehicle)) return;
		this.add(vehicle);
	}
	
}

File: trunk/NoChangeParking/src/NoChangeParking.Core/PaymentsList.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace NoChangeParking.Core
{
	public class PaymentsList:ReadOnlyCollection<Payment>
	{
		public PaymentsList():base(new List<Payment>()){/*NOOP*/}

/*		public Payment AddPayment(string vehicle, int days, Money amount) {
			if (HasPaid(vehicle)) return null;
			Payment thePayment = new Payment(vehicle, days, amount);
			Items.Add(thePayment);
			return thePayment;
		}
*/
		public bool HasPaid(string vehicle) {
			return Contains(new Payment(vehicle,null)); 
		}

		internal Payment Add(Payment payment) {
			Items.Add(payment);
			return payment;
		}
	}
}
